cli/internal/globby/globby.go

```diff
@@ -101,7 +101,10 @@ func globWalkFs(fsys iofs.FS, fsysRoot string, basePath string, includePatterns
 		// This will not error as it follows the call to checkRelativePath.
 		iofsRelativePath, _ := fs.IofsRelativePath(fsysRoot, excludePath)
 
-		// Excludes operate on entire folders.
+		// In case this is a file pattern and not a directory, add the exact pattern.
+		processedExcludes = append(processedExcludes, iofsRelativePath)
+		// TODO: we need to either document or change this behavior
+		// Excludes operate on entire folders, so we also exclude everything under this in case it represents a directory
 		processedExcludes = append(processedExcludes, filepath.Join(iofsRelativePath, "**"))
```

Comment:
Agree with you, it would be great to have one unified globbing syntax. However since this is a behavior change we should probably special case it so we can still support the existing behavior, and then move to deprecate that fully in a future version.